scripts/github-pull-requests.coffee

```
@@ -23,103 +23,148 @@
+    minutes = Math.floor(difference / 1000 / 60) - 24 * 60 * days - 60 * hours
+    hasValidTime = "#{ days }#{ hours }#{ minutes }" isnt ''
+    timeString = if hasValidTime then "#{ pluralize days, 'day' } #{ pluralize hours, 'hour' } #{ pluralize minutes, 'minute' } ago".trim() else 'just now'
+
+    "last updated #{ timeString }"
+
+  number = pr.number
+  url = pr.html_url
+  title = pr.title
+  updatedAt = pr.updated_at
+  user = pr.user.login
+  people = user
+
+  if !digest['Unassigned']
+    digest['Unassigned'] = []
+
+  if pr.assignee
+    people = "#{ user }->#{ pr.assignee.login }"
+    slackName = SLACK_USER_MAP[pr.assignee.login]
+    if slackName
+      if !digest[slackName]
+        digest[slackName] = []
+      digest[slackName].push ":octocat: #{ title } - #{ people } - #{ url } - #{ lastUpdated updatedAt }\n"
+  else
+    people = "#{ user } (Unassigned)"
+    digest['Unassigned'].push ":octocat: #{ title } - #{ people } - #{ url } - #{ lastUpdated updatedAt }\n"
+
+
+getAllPullRequests = (gitHub, repos, bot) ->
+  digest = {}
+
+  getPullRequests = (done, repo) ->
+    gitHubRepo = gitHub.repo repo
+    prs = []
+
+    gitHubRepo
+      .prsAsync()
+      .then (data) ->
+        prs = data[0]
+        addToDigest(pr, digest) for pr in prs
+      .then ->
+        done null, prs.length
+
+  parallelifyGetPullRequests = (repo) -> (done) -> getPullRequests(done, repo.full_name)
+
+  parallelGetPullRequests = []
+
+  for repo in repos
+    parallelGetPullRequests.push parallelifyGetPullRequests(repo)
+
+  async.parallel parallelGetPullRequests, (err, results) ->
+    for key, value of digest
+      if key != 'Unassigned'
+        prDashboard = "#{value.length} open pull request reviews for #{key}:\n"
+        for i of value
+          prDashboard += value[i]
+        try
+          bot.send {room: key}, prDashboard
+        catch error
+          console.log("Unable to send PR reminder to user #{key}." )
```

It might be helpful to log this error where it would be visible. If the bot breaks and stops logging, how will we be alerted? If we decide to swallow this error and make it silent, add a comment as to why.